src/utils.rs

```diff
@@ -0,0 +1,54 @@
+#[cfg(test)]
+use core::fmt::{self, Write};
+
+/// Sink that will return an error when bytes are written to it that do not match `expected`.
+#[cfg(test)]
+#[allow(dead_code)]
+pub(crate) struct WriteCompare<'a> {
+    expected: &'a str,
+    remainder: &'a str,
+}
+
+#[cfg(test)]
+impl<'a> WriteCompare<'a> {
+    pub(crate) fn new(expected: &'a str) -> Self {
+        Self { expected, remainder: expected }
+    }
+}
+
+#[cfg(test)]
+impl<'a> Write for WriteCompare<'a> {
```

Comment:
This also needs a docstring explanation as discussed below for `assert_display_eq`.